extensions/iceberg/src/main/java/io/deephaven/iceberg/util/IcebergUpdateMode.java

```diff
@@ -0,0 +1,57 @@
+//
+// Copyright (c) 2016-2024 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.iceberg.util;
+
+import io.deephaven.annotations.BuildableStyle;
+import org.immutables.value.Value;
+
+@Value.Immutable
+@BuildableStyle
+public abstract class IcebergUpdateMode {
+    private static final long REFRESH_INTERVAL_MS = 60_000L;
+
+    public static final IcebergUpdateMode STATIC = builder().updateType(IcebergUpdateType.STATIC).build();
+    @SuppressWarnings("unused")
+    public static final IcebergUpdateMode AUTO_REFRESHING =
+            builder().updateType(IcebergUpdateType.AUTO_REFRESHING).build();
+    @SuppressWarnings("unused")
+    public static final IcebergUpdateMode MANUAL_REFRESHING =
+            builder().updateType(IcebergUpdateType.MANUAL_REFRESHING).build();
+
+    public enum IcebergUpdateType {
+        STATIC, AUTO_REFRESHING, MANUAL_REFRESHING
+    }
+
+    public static Builder builder() {
+        return ImmutableIcebergUpdateMode.builder();
+    }
+
+    @SuppressWarnings("unused")
+    public static IcebergUpdateMode autoRefreshing(final long refreshMs) {
+        return ImmutableIcebergUpdateMode.builder()
+                .updateType(IcebergUpdateType.AUTO_REFRESHING)
+                .autoRefreshMs(refreshMs)
+                .build();
+    }
+
+    @Value.Default
+    public IcebergUpdateType updateType() {
+        return IcebergUpdateType.STATIC;
+    }
+
+    @Value.Default
+    public long autoRefreshMs() {
+        return REFRESH_INTERVAL_MS;
+    }
+
+    public interface Builder {
```

Comment:
I suspect we may want to pass initial snapshot (optional for static, mandatory for manual, not allowed for automatic).

Reply:
I'm not convinced. It seems natural that omitting a snapshot for a manually refreshing table would load the latest snapshot.

Reply:
My concern here is chiefly that we expect them to supply ascending snapshots, and without them supplying the initial one, there may be some difficulty in doing so.

Reply:
I understand that there's a manual "maybe update to latest", so I retract this comment as long as there *is* a way to specify the initial snapshot.